### PR TITLE
ENH GH11978 access pd.plotting._misc from plot accessor

### LINKED ISSUE #11978: ENH: other plotting tools via plot accessor
- [ ] `scatter_matrix` (deprecate, redirect to `seaborn`)
- [ ] `andrews_curve`
- [ ] `parallel_coordinates`
- [ ] `lag_plot` (maybe rename to `lag`)
- [ ] `autocorrelation_plot` (maybe rename to `autocorrelation`?)
- [ ] `bootstrap_plot` (maybe rename to `bootstrap`?)
- [ ] `radviz`

I think it would be nice to allow [scatter_matrix](http://pandas.pydata.org/pandas-docs/version/0.15.0/visualization.html#scatter-matrix-plot) to be called directly on a DataFrame. Currently, `scatter_matrix` is a separate function that takes a DataFrame as a parameter, but it seems like it would be easy enough to rework to allow it to be called directly on a DataFrame as well. Effectively, the convenience function would look something like:

``` python
def scatter_matrix(self, ...):
    pandas.tools.plotting.scatter_matrix(self.data, ...)
```

Is this feature feasible?

### PR DESCRIPTION
**Pull request by JustinZhengBC**
- [X] closes #11978
- [X] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] whatsnew entry

Allows plot accessors to use plotting functions from `pd.plotting`

- Currently only allows function calls (e.g. `df.plot.scatter_matrix()` but not `df.plot(kind='scatter_matrix')` because the handling of the `kind` keyword is interpreted with plot classes whereas the plotting tools in `pd.plotting` are functions. Should the new plotting methods be hard-coded in? Or the misc tools could be refactored into classes?

- Good tests already exist in `test_misc.py`, but they're not easily parametrizable with the plot accessor case (misc functions have frame/series parameter, plot accessor functions don't). Is it okay to have mostly copy-pasted variations of those test cases in `test_frame.py` and `test_series.py` or should they be tested together some other way?

### CODE REVIEW

File: doc/source/whatsnew/v0.24.0.rst [hunk 1]
```diff
@@ -293,6 +293,7 @@ Other Enhancements
 - :meth:`MultiIndex.to_flat_index` has been added to flatten multiple levels into a single-level :class:`Index` object.
 - :meth:`DataFrame.to_stata` and :class:` pandas.io.stata.StataWriter117` can write mixed sting columns to Stata strl format (:issue:`23633`)
 - :meth:`DataFrame.between_time` and :meth:`DataFrame.at_time` have gained the an ``axis`` parameter (:issue: `8839`)
+- many plots in the ``pandas.plotting`` module are now accessible from calling :meth:`DataFrame.plot` (:issue:`11978`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v0.24.0.rst]: Can you list the plots that are now available instead of just saying `many plots`?

Comment (JustinZhengBC) [reply doc/source/whatsnew/v0.24.0.rst]: Done

**Comment by jreback** [comment]
@JustinZhengBC can you merge master

**Comment by jreback** [comment]
thanks @JustinZhengBC

**Comment by jorisvandenbossche** [comment]
@JustinZhengBC I reverted this in #24912. And to be clear it has certainly nothing to do with the quality of the PR or so, just that after consideration decided that we would rather not expose those extra plotting methods. And sorry for the quick PR, it's just that I wanted to do this before the release which hopefully happens today. In any case, thanks for your contributions!

**Pull request merged**